src/arrayMethodJoin.js

```diff
@@ -5,8 +5,26 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let newSeparator = separator;
+
+    if (separator === undefined) {
+      newSeparator = ',';
+    }
+
+    let list = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== undefined && this[i] !== null) {
```

Comment:
You can use:
```js
 if (this[i] != null) {
 ```
 The logic here is that
 ```js
 (null == undefined) === true //  and shortens the code a little

Reply:
thank you, Umut, but when I use this[i] != null it returns an error to me, saying that I should use  !== null